### ISSUE TITLE
Weatherflow integration is using deprecated forecast

### ISSUE THREAD

**Issue by mtwhitley** [briis/hass-weatherflow#87]
As of Home Assistant v2023.9.0 the following warning is given:

<img width="582" alt="Screenshot 2023-09-08 at 12 09 45 PM" src="https://github.com/briis/hass-weatherflow/assets/14807414/c361b675-e92a-4414-b359-9fd05db6b5ec">

I'm sure you are already aware, but thought I would open a issue anyway.

**Comment by briis** [comment]
I know. There is still 6 month to fix this, and I am working on it.

**Comment by Redmage753** [comment]
Came here to add to this, glad to see it already acknowledged! Because I found this helpful from another weather integration, going to include this here as a reference:
https://developers.home-assistant.io/blog/2023/08/07/weather_entity_forecast_types/

**Comment by briis** [comment]
I was justing looking through the coming 2023.10 release, and I can see that there now will be a `core` integration called `weatherflow`. This will use the UDP API, and as such will contain a less amount of sensors, and it will also not include any Forecast data, as this **must** be retrieved using the REST API.

What the future of this complete integration will be I have not decided yet, but the domain name now seems taken by a `core` integration. 

What I have done, is that I have now created a first test version of a Forecast only version of WeatherFlow that fully supports the new Forecast format. If you want to test it out, go to [this Github](https://github.com/briis/weatherflow_forecast) page and follow the instructions on how to add it to HACS. This is currently only been tested by me, so if any one else wants to give it a spin, feel free to do so.

Eventually a combination of the `core` weatherflow integration and this new Forecast integration, might be ideal?

**Comment by miwhitle** [comment]
Thanks for keeping us in the loop. Will check out the new forecast integration, thank you!

Side note: I just noticed your name. Are you the cycling legend or just happen to have the same name and get this question all the time? (Sorry if it's the latter). :)

**Comment by drothenberger** [comment]
@briis: Would it be possible to add sensors for the other values that are only available through the REST API, such as NC Rain?

**Comment by briis** [comment]
> Are you the cycling legend or just happen to have the same name and get this question all the time? (Sorry if it's the latter). :)

🚴‍♂️ No I am not the cycling legend, even though I actually do spend a significant amount of time on the Racing bike. 😎

**Comment by briis** [comment]
> Would it be possible to add sensors for the other values that are only available through the REST API, such as NC Rain?

It will not be difficult to do, I just need to consider if I want to do a full migration of this Integration or not.

**Comment by mtwhitley** [comment]
> > Are you the cycling legend or just happen to have the same name and get this question all the time? (Sorry if it's the latter). :)
> 
> 🚴‍♂️ No I am not the cycling legend, even though I actually do spend a significant amount of time on the Racing bike. 😎

Awesome! Thanks for this integration and for satisfying my curiosity! ❤️

**Comment by Klarstein** [comment]
> > Would it be possible to add sensors for the other values that are only available through the REST API, such as NC Rain?
> 
> It will not be difficult to do, I just need to consider if I want to do a full migration of this Integration or not.

Thank you for your effort so far! It has been rock solid. Will the (your) current plugin still work with 2023.10 if I don't use the core Weatherflow integration? Or will it generate heaps of errors since the domain is meant to be used with the core integration?

**Comment by briis** [comment]
> Will the (your) current plugin still work with 2023.10 if I don't use the core Weatherflow integration?

As far as my testing shows, if there is an integration in the `custom_components` directory with the same name as a `core` integration, the Custom Component will be used. I have not done extensive testing on that, but it seams so.
Short answer: I do believe it will still work - but no guarantees.

**Comment by matthewhadley** [comment]
I can confirm that custom integrations with the same namespace as core integrations will get loaded in place of (and so "replace") the core equivalent. This has been the way to sometimes "patch" core integrations while waiting for official updates/fixes for any issues.

**Comment by Aipoc76** [comment]
just rename your RealWeatherFlow :)

**Comment by briis** [comment]
Yes I could just rename it, but I still had to to some work to get it running with the new Forecast specifications.
I have decided to add the sensor data to the new Integration, but making it optional to enable, so that you can have forecast and sensor data, or just forecast.
My new API Wrapper is now done and supporting this, so now I need to modify the new Custom component.

**Comment by briis** [comment]
> So, if we own a weather station and want to access local sensor data as well as the forecast, would we then use the new core intergation and weatherflow_forecast together or would we use weatherflowmqtt and weatherflow_forecast?

Yes to both - eventually. 
Once WeatherFlow Forecast is complete, you will have the option of adding local sensors also, that will bring the missing sensors you cannot get from the UDP API, which the new `weatherflow` integration uses.

Currently WeatherFlow2MQTT also has the forecast option built-in, but I will remove that, so that I do not have to maintain this in two places. So the a Combination of WeatherFlow2MQTT and WeatherFlow forecast without local sensors, will give you all the sensors, plus the forecast.

**Comment by briis** [comment]
> Also, how could we get the is-raining binary sensor back if using weatherflowmqtt?

Simply create a Template Binary sensor, that checks if Rain Rate is > 0

**Comment by jgosnell56** [comment]
> Yes to both - eventually.
Once WeatherFlow Forecast is complete, you will have the option of adding local sensors also, that will bring the missing sensors you cannot get from the UDP API, which the new weatherflow integration uses.
So if I am reading this correctly, once complete the Weatherflow Foecast is complete, it is the only integration we will need for local sensor access as well as the forecast?  Any idea what the timeline may be for that?

**Comment by drothenberger** [comment]
> So if I am reading this correctly, once complete the Weatherflow Foecast is the only integration we will need for local sensor access as well as the forecast?  

The new core `weatherflow` integration will provide local sensor access. The Weatherflow Forecast integration will provide the access through the REST API, providing both the weather forecast and the other sensors that are not available through the UDP interface. 

So, you will need both integrations.

**Comment by PhillySports** [comment]
Thank you for this project and all your time on it.   If we are using your custom integration, should we uninstall that one go with the core integration and then install the new Forecast Weatherflow integration.   Is that what you are recommending?

**Comment by drothenberger** [comment]
>If we are using your custom integration, should we uninstall that one go with the core integration and then install the new Forecast Weatherflow integration.   Is that what you are recommending?

I'm not the code owner, but I believe that is the long-term recommendation.

Currently, the core integration doesn't provide sensors for everything, including rain. Eventually, the WeatherFlow Forecast integration will include the sensor values from the REST API (including rain). Until then, I am personally sticking with this integration and the Weatherflow2MQTT add-on.

**Comment by ErikSGross** [comment]
Thanks @briis for all the previous work on this project and for the new Forecast integration.  I just installed the Forecast integration, and it worked perfectly the first time. 

I haven't updated to HA 2023.10, but I think I'll keep the existing `custom_component` until there's an option to get the REST sensors that are missing from the new core integration.  

I'm assuming one of the REST-only sensors is the "Checked" version of the precipitation values - my understanding is WeatherFlow uses their cloud data from other stations/services, compares it against your raw readings from your device, and then provides what they believe is a more accurate precipitation value.   I've only had my Tempest for about a month, but what I'm seeing is that the "Checked" values are closer to what I see with a traditional tube-style rain gauge in my yard.  

Hopefully the "Checked" versions of sensors can be part of the new Forecast integration, even if they aren't really a "forecast".  I think the same applies for lightning strikes, at least from what I've heard from a friend who lives in an area where they actually get lightning more than once a year.

**Comment by briis** [comment]
> Hopefully the "Checked" versions of sensors can be part of the new Forecast integration, even if they aren't really a "forecast". I think the same applies for lightning strikes, at least from what I've heard from a friend who lives in an area where they actually get lightning more than once a year.

The *checked* version applies to the Precipitation, and is a US only thing. I will get it included - cannot test it, as I do not get these sensors here where I live.

If all goes well, a test release of **WeatherFlow Forecast** should come out sunday, with support for Sensors.
